uaclient/version.py

```diff
@@ -0,0 +1,23 @@
+"""
+Version determination functions
+
+These are in their own file so they can be imported by setup.py before we have
+any of our dependencies installed.
+"""
+import os.path
+from subprocess import check_output
+
+
+__VERSION__ = '19.1'
+PACKAGED_VERSION = '@@PACKAGED_VERSION@@'
+
+
+def get_version(_args=None):
+    """Return the package version if set, otherwise return git describe."""
+    if not PACKAGED_VERSION.startswith('@@PACKAGED_VERSION'):
+        return PACKAGED_VERSION
+    topdir = os.path.dirname(os.path.dirname(__file__))
+    if os.path.exists(os.path.join(topdir, '.git')):
+        cmd = ['git', 'describe', '--abbrev=8', '--match=[0-9]*', '--long']
+        return check_output(cmd, text=True, encoding='utf-8').strip()
```

Comment:
Sent you a message on IRC, but the text=True is not something I am familiar with and was giving me traceback: https://paste.ubuntu.com/p/SbcmY8PGmV/

If I remove the text=True, I run into being unable to create /etc/... files, but that at least moved me on.